src/main/java/uk/nhs/tis/sync/job/PostFundingStatusSyncJob.java

```diff
@@ -0,0 +1,140 @@
+ * status based on specified criteria.
+ */
+@Component
+@ManagedResource(objectName = "sync.mbean:name=PostFundingStatusSyncJob",
+    description = "Job for updating funding status for posts")
+@Slf4j
+public class PostFundingStatusSyncJob extends CommonSyncJobTemplate<Post> {
+
+  private static final int FIFTEEN_MIN = 15 * 60 * 1000;
+  private static final int DEFAULT_PAGE_SIZE = 5000;
+  private static final String BASE_QUERY = " SELECT postId "
+      + "  FROM PostFunding "
+      + "      WHERE postId > :lastPostId "
+      + "      AND startDate IS NOT NULL "
+      + "      AND (endDate = ':endDate' OR endDate IS NULL) "
+      + " GROUP BY postId "
+      + " ORDER BY postId LIMIT :pageSize ";
+
+  public PostFundingStatusSyncJob(EntityManagerFactory entityManagerFactory,
+      @Autowired(required = false) ApplicationEventPublisher applicationEventPublisher) {
+    super(entityManagerFactory, applicationEventPublisher);
+  }
+
+  @Scheduled(cron = "${application.cron.postFundingStatusSyncJob}")
+  @SchedulerLock(name = "postFundingStatusSyncJobTask", lockAtLeastFor = FIFTEEN_MIN,
+      lockAtMostFor = FIFTEEN_MIN)
+  @ManagedOperation(description = "Run sync of modifying the post funding status")
+  public void postFundingStatusSyncJob() {
+    runSyncJob(null);
+  }
+
+  public void run(String params) {
+    runSyncJob(params);
+  }
+
+  protected String buildQueryForDate() {
+    LocalDate today = LocalDate.now();
+    String endDate = today.minusDays(1).format(DateTimeFormatter.ISO_LOCAL_DATE);
+    return BASE_QUERY.replace(":endDate", endDate).replace(":pageSize", "" + DEFAULT_PAGE_SIZE);
+  }
+
+  @Override
+  protected String assembleQueryString(String option) {
+    String queryString = buildQueryForDate();
+    log.debug("Job will run with query:[{}]", queryString);
+    return queryString;
+  }
+
+  @Override
+  protected List<EntityData> collectData(Map<String, Long> ids, String queryString,
+      EntityManager entityManager) {
+    long lastPostId = ids.get(LAST_ENTITY_ID);
+    Query query =
+        entityManager.createNativeQuery(queryString).setParameter("lastPostId", lastPostId);
+
+    List<BigInteger> resultList = query.getResultList();
+    return resultList.stream().filter(Objects::nonNull).map(objArr ->
+        new EntityData().entityId(objArr.longValue())
+    ).collect(Collectors.toList());
+  }
+
+  @Override
+  protected int convertData(Set<Post> entitiesToSave, List<EntityData> entityData,
+      EntityManager entityManager) {
+    int entities = entityData.size();
+    for (EntityData entity : entityData) {
+      Post post = entityManager.find(Post.class, entity.getEntityId());
+
+      if (post != null) {
+        Set<PostFunding> fundings = post.getFundings();
+
+        if (fundings.size() > 1) {
+          entitiesToSave.add(post);
+        } else if (fundings.size() == 1) {
+          post.setFundingStatus(Status.INACTIVE);
+          entitiesToSave.add(post);
+        }
```

Comment:
This is a crucial bit of code and I think there are cases that aren't tested.  Given at least 1 funding must have at least 1 funding that ended 'yesterday'.
A post which was `CURRENT` & with 2 fundings that ended yesterday. I expect that we would want it to become `INACTIVE`.

Reply:
The logic has been tested locally

Reply:
Code has been refactored and  a test has been written for this.